### PR TITLE
feat: atclient pkam_authenticate

### PR DESCRIPTION
**Pull request by JeremyTubongbanua**
related to #39 

**- What I did**

__general changes__:
- use #ifdef X_H, #define X_H, #endif as opposed to #pragma once to ensure compatibility with different compilers
- utilize unsigned long as opposed to size_t to ensure compatibility with different compilers
- run_ctest.sh in atclient and atchops to run ctests

__atchops__:

- new aes.h and aes.c -> contains generic aes key generation and aes bits (128 and 256) enum
- new iv.h and iv.c -> contains generation of base64 and non-base64 initialization vectors
- new rsakey.h -> out sourced the rsakey interactions from the rsa lib
- refactored tests to print things more tidely

__atclient__:

- new atclient_pkam_authenticate function:

```c 
int atclient_pkam_authenticate(atclient_ctx *ctx, atclient_atkeys atkeys, const char *atsign);
```

- new atkey.h and atkey.c -> represents an AtKey in the atProtocol
- new atkeys.h and atkeys.c -> holds decrypted RSA and AES keys in base64 and atchops keys to be used in subsequent cryptography
- new atkeysfile.h and atkeys.c -> represents the atkeys file (holds encrypted cryptographic keys and base64) as well as file reading functionality
- new atlogger.h and atlogger.c -> logging utility
- new atstr.h and atstr.c -> represents the 3 dimensions of a string: buffer length, buffer pointer, and the written length. to be used inside of structs
- new metadata.h and metadata.c -> represents metadata in the atProtocol

__examples__:

- examples refactoring - upload placeholders for include folders and static libraries and not the actual files
- new atkey example
- new pkam_authenticate example
- new reading_atkeys example

**- How I did it**

with my bare hands

**- How to verify it**

tests (even though refactored) are passing

**- Description for the changelog**

### CODE REVIEW

File: examples/pkam_authenticate/src/main.c [hunk 1]
```diff
@@ -0,0 +1,68 @@
+#include <stdio.h>
+#include <atclient/atclient.h>
+#include <atclient/atkeysfile.h>
+#include <atclient/atlogger.h>
+
+#define ROOT_HOST "root.atsign.org"
+#define ROOT_PORT 64
+
+#define ATKEYSFILE_PATH "/Users/jeremytubongbanua/.atsign/keys/@smoothalligator_key.atKeys"
```
Comment (xavierchanth) [inline examples/pkam_authenticate/src/main.c:9]: Careful of hard-coded values maybe add a define.h (gitignored) and a define-template.h (with empty values) or however you want to do it

**Review by realvarx — APPROVED** [review]
LGTM

**Comment by JeremyTubongbanua** [comment]
@realvarx introduced #46

File: examples/pkam_authenticate/src/main.c (hunk 1 above)
Comment (JeremyTubongbanua) [reply examples/pkam_authenticate/src/main.c:9]: This is in examples. Should I still do this?

Comment (xavierchanth) [reply examples/pkam_authenticate/src/main.c:9]: Meh, don't let it stop this PR from being merged. But consider changing in the future

Comment (JeremyTubongbanua) [reply examples/pkam_authenticate/src/main.c:9]: Ok

**Pull request merged**